_posts/2021-03-02-introducing-new-window-features.md

```diff
@@ -0,0 +1,317 @@
+includes all rows until the partition bound. `CURRENT ROW` is more tricky. It
+includes all rows where values are the same as in the current row. We call
+them a _peer group_.
+
+It's time to introduce the first of Trino's new additions:
+
+### <a name="new features"/> Full support for frame type RANGE
+
+Since version 346, it is possible to specify `RANGE` with an offset value. The
+frame includes all rows whose value is within this range from the current row.
+
+Let's modify our example:
+
+```
+SELECT avg(totalprice) OVER (PARTITION BY custkey ORDER BY orderdate
+   RANGE BETWEEN interval '1' month PRECEDING AND CURRENT ROW)
+FROM orders
+```
+
+Now, we don't always get the average price since the beginning, but the average
+price from the preceding month of the current entry. Note that the offset
+`interval '1' month` applies to `orderdate`, which is the sorting column.
+
+Of course, we don't have to order by date. The sorting column can be of any
+numeric or date/time type, and the offset must be compatible. Also, the offset
+doesn't have to be a literal. It can come in another column of a table or,
+generally, it can be any expression, as long as the type matches.
+
+Frame of type `RANGE` does not quite fit in the abstraction of a “sliding
+window”. Frames can be bigger or smaller depending not only on the offset
+values but also on the actual input data. A long series of similar entries can
+produce a huge frame, while a gap in input values can result in an empty frame.
+
+For illustration, imagine a group of students, and the results of some test they
+took. Our table has two columns: `student_id` and `result`, which is the number
+of points. For each student, let's find how many students did better by 1 to 2
+points:
+
+```
+WITH students_results(student_id, result) AS (VALUES
+    ('student_1', 18),
+    ('student_2', 16),
+    ('student_3', 18),
+    ('student_4', 18),
+    ('student_5', 10),
+    ('student_6', 20),
+    ('student_7', 16))
+SELECT student_id, result, count(*) OVER (ORDER BY result
+   RANGE BETWEEN 1 FOLLOWING AND 2 FOLLOWING) AS close_better_scores_count
+FROM students_results;
+
+ student_id | result | close_better_scores_count
+------------+--------+---------------------------
+ student_5  |     10 |                         0
+ student_2  |     16 |                         3
+ student_7  |     16 |                         3
+ student_1  |     18 |                         1
+ student_3  |     18 |                         1
+ student_4  |     18 |                         1
+ student_6  |     20 |                         0
+(7 rows)
+```
+
+Note that the frame does not contain the current row. For a particular student,
+it only includes students with better results, and not themselves. For the
+unfortunate `student_5`, there are no students with similar test results.
+Also, for the lucky `student_6` who scored the most points.
+
+Back to the internal stuff: how do we deal with finding the frame bounds with
```

Comment:
Why "Back"? It didn't seem like we were previously discussing implementation details. Also, I might put this in a separate section that talks about what happens under the covers, to separate it from "how do window functions work"